File: MSRtail_dep_tStudent/MSRtail_dep_tStudent.m

close all
clear
clc

rho = -0.8:0.2:0.8;
n   = length(rho);
DoF = 3;

for i = 1:n
    v = 0.5:0.01:1;
    lambda = 2*(1-tcdf(sqrt((DoF+1)./(DoF./(tinv(v,3).^2)+1))*sqrt((1-rho(i))/(1+rho(i))),DoF+1));

%lambda
hold on
plot(v,lambda,'k');
hold off
text(0.54,0.4,'rho = -0.8')
text(0.79,0.7,'rho = 0.6')
text(0.86,0.75,'rho = 0.8')
xlabel('v')
ylabel('lambda_U(v)')
end

File: MSRtail_dep_tStudent/MSRtail_dep_tStudent.R

rm(list = ls(all = TRUE))
#setwd("C:/...")

rho = seq(-0.8, 0.8, by=0.2)
n   = length(rho)
v   = seq(0.5,1,by=0.01)
DoF = 3

lambda = matrix(1,nrow=length(v),ncol=length(rho))
for (i in 1:n){
	lambda[, i] = 2*(1-pt(sqrt((DoF+1)/(DoF/(qt(v,3)^2)+1))*
        sqrt((1-rho[i])/(1+rho[i])),DoF+1))
}
	
plot(v,  lambda[,1], type = "l", xlab = "v", ylab = expression(lambda[U](v)))
lines(v, lambda[,2])
lines(v, lambda[,3])
lines(v, lambda[,4])
lines(v, lambda[,5])
lines(v, lambda[,6])
lines(v, lambda[,7])
lines(v, lambda[,8])
lines(v, lambda[,9])
text(0.55,0.4,"rho = -0.8")
text(0.81,0.7,"rho = 0.6")
text(0.88,0.75,"rho = 0.8")